### PR TITLE
Resolves Issue518 - Don't overwrite modal-inner when classname is passed

### PR DESCRIPTION
**Pull request by Amihozar**
Resolving #518 by customizing Modal className property inspired by [React Modal buildCalssName](https://github.com/reactjs/react-modal/blob/702ff620924af916cf08223871d563efd5a96294/src/components/ModalPortal.js#L331)

## Author Checklist

- [x] Add unit test(s)
- [ ] Update version in package.json (see the [versioning guidelines](https://github.com/LaunchPadLab/opex-public/blob/master/gists/npm-package-guidelines.md#pull-requests-and-deployments))
- [ ] Update documentation (if necessary)
- [ ] Add story to storybook (if necessary)
- [ ] Assign dev reviewer

### CODE REVIEW

File: src/modal.js [hunk 1]
```diff
@@ -55,15 +56,28 @@ function getRootElement() {
...
+  // No className specefied or className property is empty
+  if (!additional)
+    return main
+  // A nice and clean className, nothing special to do
+  if (typeof additional === 'string')
+    return `${main} ${additional}`
+  // react-modal will handle it
+  // see http://reactcommunity.org/react-modal/styles/classes/
+  if (typeof additional === 'object')
+    return additional
```
Comment (chawes13) [inline src/modal.js:69]: Would this override `main`?

**Comment by chawes13** [comment]
@Amihozar Thanks for the PR! If you use the word "resolve", "resolves", or "resolved", then the issue will be linked and automatically close when this PR is merged. However, "resolving" is not one of those key words. More info here: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

File: src/modal.js (hunk 1 above)
Comment (Amihozar) [reply src/modal.js:69]: > 

Yes it does.
Passing an object to className will override both main and ReactModal__Content classes.

**Review by chawes13 — CHANGES_REQUESTED** [review]
Looking great! Can you update your base branch to `LaunchPadLab:v6` and update the migration guide accordingly? There might be apps that rely on the behavior to overwrite the class, so I don't think we should include it in `v5` (i.e., the current major version)

**Comment by chawes13** [comment]
This PR is deprecated by https://github.com/LaunchPadLab/lp-components/pull/554

**Pull request closed without merging**